erpnext/crm/doctype/email_campaign/email_campaign.py

```
@@ -77,58 +80,129 @@ def update_status(self):
+			continue
+
+		# Find schedules that match today
 		for entry in campaign.get("campaign_schedules"):
-			scheduled_date = add_days(email_campaign.get("start_date"), entry.get("send_after_days"))
-			if scheduled_date == getdate(today()):
-				send_mail(entry, email_campaign)
+			try:
+				scheduled_date = add_days(getdate(email_campaign.start_date), entry.get("send_after_days"))
+				if scheduled_date == today_date:
+					send_mail(entry, email_campaign)
+			except Exception:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)